src/scripts/main.js

```diff
@@ -1 +1,88 @@
 'use strict';
+
+let bodyDoc;
+let rightClick;
+let leftClick;
+
+if (document.querySelector('body') !== null) {
+  bodyDoc = document.querySelector('body');
+}
+
+function makeDiv(className) {
+  const div = document.createElement('div');
+
+  div.setAttribute('data-qa', 'notification');
+  div.setAttribute('class', className);
+  bodyDoc.appendChild(div);
+
+  return div;
+}
+
+const firstPromise = () => {
+  return new Promise((resolve, reject) => {
+    document.addEventListener('click', () => {
+      leftClick = true;
+
+      resolve('First promise was resolved');
+    });
+
+    // eslint-disable-next-line prefer-promise-reject-errors
+    setTimeout(() => reject('First promise was rejected!'), 3000);
```

Comment:
The 'click' event listener added inside the promise will never be removed, potentially causing memory leaks and unexpected behavior if 'firstPromise' is called multiple times. Consider removing the event listener after the promise is resolved or rejected.